src/App.tsx

```diff
@@ -17,18 +53,31 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                status={completedStatus}
+                setStatus={setCompletedStatus}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+
+              {todos.length
+                ? (
+                  <TodoList
+                    todos={visibleTodos}
+                    setSelectedTodo={setSelectedTodo}
```

Comment:
don't pass function for updating state in child component, create function wrapper for it, fix in all places